sql/src/main/java/io/crate/planner/operators/LogicalPlan.java

```diff
@@ -104,6 +104,37 @@
         LogicalPlan build(TableStats tableStats, Set<Symbol> usedBeforeNextFetch);
     }
 
+    /**
+     * Tries to "push-down" a LogicalPlan to this plan or the children of
+     * this plan. A push-down is an optimization which does not change the
+     * outcome of the plan but, ideally, makes it execute more efficiently.
+     *
+     * For example, pushing down an *Order*:
+     *
+     *              *Order*                      Union
+     *                 |                        /     \
+     *                 |                       /       \
+     *               Union                 *Order*  *Order*
```

Comment:
since we merge the Order operators this needs to be adjusted with *NewOrder* for example.